src/Serilog.HttpClient/LoggingDelegatingHandler.cs

```diff
@@ -203,22 +187,26 @@ static double GetElapsedMilliseconds(long start, long stop)
                     }
                 }
 
-                var responseData = new
-                {
-                    StatusCode = (int?)resp?.StatusCode,
-                    IsSucceed = isRequestOk,
-                    ElapsedMilliseconds = elapsedMs,
-                    BodyString = responseBodyText,
-                    Body = responseBody,
-                    Header = responseHeader,
-                };
-
-                _logger.Write(level, ex, _options.MessageTemplate, new
+                using (Serilog.Context.LogContext.PushProperty("RequestMethod", req.Method))
+                using (Serilog.Context.LogContext.PushProperty("RequestUri", req.RequestUri))
+                using (Serilog.Context.LogContext.PushProperty("RequestScheme", req.RequestUri.Scheme))
+                using (Serilog.Context.LogContext.PushProperty("RequestHost", req.RequestUri.Host))
+                using (Serilog.Context.LogContext.PushProperty("RequestPath", req.RequestUri.AbsolutePath))
+                using (Serilog.Context.LogContext.PushProperty("RequestQueryString", req.RequestUri.Query))
+                using (Serilog.Context.LogContext.PushProperty("RequestQuery", requestQuery))
+                using (Serilog.Context.LogContext.PushProperty("RequestBodyString", requestBodyText))
+                using (Serilog.Context.LogContext.PushProperty("RequestBody", requestBody, destructureObjects: true))
+                using (Serilog.Context.LogContext.PushProperty("RequestHeaders", requestHeaders))
+                using (Serilog.Context.LogContext.PushProperty("StatusCode", resp?.StatusCode.ToString()))
+                using (Serilog.Context.LogContext.PushProperty("IsSucceed", isRequestOk))
+                using (Serilog.Context.LogContext.PushProperty("ElapsedMilliseconds", elapsedMs))
+                using (Serilog.Context.LogContext.PushProperty("ResponseBodyString", responseBodyText))
+                using (Serilog.Context.LogContext.PushProperty("ResponseBody", responseBody, destructureObjects: true))
+                using (Serilog.Context.LogContext.PushProperty("ResponseHeaders", responseHeaders))
```

Comment:
This approach results in a named properties being added to the log event, so you can refer to them in the message template by name, and they're stored within the log event as well. Previously they were all bundled under a Context object that prevented accessing the individual properties directly.

Reply:
Bundling into one property allowed me to search and navigate easier on tons of logs with many properties to watch. as I implemented this in brother projects: [Serilog.AspNetCore.Plus](https://github.com/alirezavafi/serilog-aspnetcore-plus) and [AutoWrapper.AspNetCore.Plus](https://github.com/alirezavafi/autowrapper-aspnetcore-plus). But your comment and approch are valid and acceptable and made me in a doubt.